network/src/noise/handshake.rs

```diff
@@ -145,9 +151,13 @@ pub struct NoiseUpgrader {
 
 impl NoiseUpgrader {
     /// Create a new NoiseConfig with the provided keypair and authentication mode.
-    pub fn new(peer_id: PeerId, key: x25519::PrivateKey, auth_mode: HandshakeAuthMode) -> Self {
+    pub fn new(
+        network_context: Arc<NetworkContext>,
+        key: x25519::PrivateKey,
+        auth_mode: HandshakeAuthMode,
```

Comment:
shouldn't this be part of NetworkContext?

Reply:
auth_mode? I think a lot could be part of network context... worth reviewing at some point in time what other values we may potentially want... for example, I wondered why chain id wasn't part of it

Reply:
yeah the auth_mode

Reply:
I'll make a note of that : o compiling some notes about network